gremlin-go/cucumber/featureSteps_test.go

```diff
@@ -361,22 +431,216 @@ func (tg *tinkerPopGraph) theResultShouldBe(characterizedAs string, table *godog
 	}
 }
 
-func contains(list []interface{}, item interface{}) bool {
-	for _, v := range list {
-		if reflect.DeepEqual(v, item) {
-			return true
+func compareMapEquals(expected map[interface{}]interface{}, actual map[interface{}]interface{}) bool {
+	for k, a := range actual {
+		var e interface{}
+		containsKey := false
+		for ke, ee := range expected {
+			if fmt.Sprint(k) == fmt.Sprint(ke) {
+				containsKey = true
+				e = ee
+				break
+			} else {
+				if reflect.ValueOf(k).Kind() == reflect.Ptr &&
+					reflect.ValueOf(ke).Kind() == reflect.Ptr {
+					switch k.(type) {
+					case *gremlingo.Vertex:
+						switch ke.(type) {
+						case *gremlingo.Vertex:
+							if fmt.Sprint(*k.(*gremlingo.Vertex)) == fmt.Sprint(*ke.(*gremlingo.Vertex)) {
+								containsKey = true
+							}
+						default:
+							// Not equal.
+						}
+					default:
+						// If we are here we probably need to implement an additional type like the Vertex above.
+						if fmt.Sprint(*k.(*interface{})) == fmt.Sprint(*ke.(*interface{})) {
+							fmt.Println("WARNING: Encountered unknown pointer type as map key.")
+							containsKey = true
+						}
+					}
+					if containsKey {
+						e = ee
+						break
+					}
+				}
+			}
+		}
+		if !containsKey {
+			fmt.Printf("Map comparison error: Failed to find key %s in %v\n", k, expected)
+			return false
+		}
+
+		if a == nil && e == nil {
+			continue
+		} else if a == nil || e == nil {
+			// One value is nil, other is not. They are not equal.
+			fmt.Printf("Map comparison error: One map has a nil key, other does not.\n")
+			return false
+		} else {
+			switch reflect.TypeOf(a).Kind() {
+			case reflect.Array, reflect.Slice:
+				switch reflect.TypeOf(e).Kind() {
+				case reflect.Array, reflect.Slice:
+					// Compare arrays
+					if !compareListEqualsWithoutOrder(a.([]interface{}), e.([]interface{})) {
+						return false
+					}
+				default:
+					fmt.Printf("Map comparison error: Expected type is Array/Slice, actual is %s.\n", reflect.TypeOf(a).Kind())
+					return false
+				}
+			case reflect.Map:
+				switch reflect.TypeOf(a).Kind() {
+				case reflect.Map:
+					// Compare maps
+					if !compareMapEquals(a.(map[interface{}]interface{}), e.(map[interface{}]interface{})) {
+						return false
+					}
+				default:
+					fmt.Printf("Map comparison error: Expected type is Map, actual is %s.\n", reflect.TypeOf(a).Kind())
+					return false
+				}
+			default:
+				if fmt.Sprint(a) != fmt.Sprint(e) {
+					fmt.Printf("Map comparison error: Expected != Actual (%s!=%s)\n", fmt.Sprint(a), fmt.Sprint(e))
+					return false
+				}
+			}
+		}
+	}
+	return true
+}
+
+func compareListEqualsWithoutOrder(expected []interface{}, actual []interface{}) bool {
```

Comment:
can we use testify library?
https://github.com/stretchr/testify/blob/master/assert/assertions.go#L899

Reply:
Not sure if it's needed at this point, since our tests are passing. It'll add extra licensing overhead.